File: WebDevProject/src/java/DataBase/StudentDB.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package DataBase;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author Justin
 */
public class StudentDB {
    
    
    public static void updateBalance(int user_id, int balance) {
        ConnectionPool pool = ConnectionPool.getInstance();
        Connection connection = pool.getConnection();
        PreparedStatement ps = null;
        String query = "UPDATE STUDENT SET BALANCE = ? WHERE STU_ID = ?";
        try {
            ps = connection.prepareStatement(query);
            ps.setInt(1, balance);
            ps.setInt(2, PrivilegeDB.getStudentIDByUserID(user_id));
            ps.executeUpdate();
        } catch (SQLException e) {
            Logger.getLogger(PrivilegeDB.class.getName()).log(Level.SEVERE, null, e);
        } finally {
            DBUtil.closePreparedStatement(ps);
            pool.freeConnection(connection);
        }
        
        
    }
    
    public static int getBalance(int user_id) {
        ConnectionPool pool = ConnectionPool.getInstance();
        Connection connection = pool.getConnection();
        PreparedStatement ps = null;
        ResultSet rs = null;
        String query = "SELECT BALANCE FROM STUDENT WHERE STU_ID = ?";
        try {
            ps = connection.prepareStatement(query);
            ps.setInt(1, PrivilegeDB.getStudentIDByUserID(user_id));
            rs = ps.executeQuery();
            if (rs.next())
                return rs.getInt("BALANCE");
        } catch (SQLException e) {
            Logger.getLogger(PrivilegeDB.class.getName()).log(Level.SEVERE, null, e);
        } finally {
            DBUtil.closePreparedStatement(ps);
            pool.freeConnection(connection);
            DBUtil.closeResultSet(rs);
        }    
        return 0;
    }
    
    public static int getCourseCreditsAndCost(int user_id) {
        ConnectionPool pool = ConnectionPool.getInstance();
        Connection connection = pool.getConnection();
        PreparedStatement ps = null;
        ResultSet rs = null;
        int cost = 0;
        String query =  "SELECT c.COURSE_COST, c.COURSE_CREDITS FROM COURSE c " +
                        "INNER JOIN SECTION s ON (c.COURSE_ID = s.COURSE_ID " +
                        "AND c.DEPT_ID = s.DEPT_ID) INNER JOIN REGISTRATION r " +
                        "ON r.SECTION_NUM = s.SECTION_NUM WHERE r.STU_ID = ?";
        try {
            ps = connection.prepareStatement(query);
            int stu_id = PrivilegeDB.getStudentIDByUserID(user_id);
            ps.setInt(1, stu_id);
            rs = ps.executeQuery();
            
            while (rs.next()) {
                cost += rs.getInt("COURSE_COST");
                cost = cost + (rs.getInt("COURSE_CREDITS") * 87);                
            }
        } catch (SQLException e) {
            Logger.getLogger(PrivilegeDB.class.getName()).log(Level.SEVERE, null, e);
        } finally {
            DBUtil.closePreparedStatement(ps);
            pool.freeConnection(connection);
            DBUtil.closeResultSet(rs);
        }    
        return cost; 
        
    }
}

File: WebDevProject_PHP/app/model/StudentDB.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of StudentDB
 *
 * @author Justin
 */
class StudentDB {
    
    public function __construct() {
        
    }
    
    public function getStudentID($user_id) {
        include('/database.php');
        $query = 'SELECT * FROM STUDENT WHERE USER_ID = :user_id';
        try {
            $statement = $db->prepare($query);
            $statement->bindValue(':user_id', $user_id);
            $statement->execute();
            $result = $statement->fetchAll();
            $statement->closeCursor();
            return $result[0]['STU_ID'];
        } catch (Exception $e) {
            $error_message = $e->getMessage();
            echo $error_message;
        }
    }
    
    public function isStudent($user_id) {
        include('/database.php');
        $query = 'SELECT COUNT(*) FROM STUDENT WHERE USER_ID = :user_id';
        try {
            $statement = $db->prepare($query);
            $statement->bindValue(':user_id', $user_id);
            $statement->execute();
            $result = $statement->fetchAll();
            $statement->closeCursor();
            return ($result[0]["COUNT(*)"] != '0');
        } catch (Exception $e) {
            $error_message = $e->getMessage();
            echo $error_message;
        }
        return false;
    }
    
    public function getBalanceID($user_id) {
        include('/database.php');
        $query = 'SELECT BALANCE FROM STUDENT WHERE USER_ID = :user_id';
        try {
            $statement = $db->prepare($query);
            $statement->bindValue(':user_id', $user_id);
            $statement->execute();
            $result = $statement->fetchAll();
            $statement->closeCursor();
            return $result[0]['BALANCE'];
        } catch (Exception $e) {
            $error_message = $e->getMessage();
            echo $error_message;
        }
    }
    
    public function setBalanceID($user_id, $balance) {
        include('/database.php');
        $query = 'UPDATE STUDENT SET BALANCE = :balance WHERE STU_ID = :stu_id';
        try {
            $statement = $db->prepare($query);
            $statement->bindValue(':balance', $balance);
            $statement->bindValue(':stu_id', $this->getStudentID($user_id));
            $statement->execute();
            $statement->closeCursor();
            return $this->getBalanceID($user_id);
        } catch (Exception $e) {
            $error_message = $e->getMessage();
            echo $error_message;
        }
        
    }
}
